packages/utils/src/lib/media/media.ts

```diff
@@ -119,17 +119,33 @@ export class MediaHelpers {
 	 * Load an image from a url.
 	 * @public
 	 */
-	static loadImage(src: string): Promise<HTMLImageElement> {
+	static getImageAndDimensions(
+		src: string
+	): Promise<{ w: number; h: number; image: HTMLImageElement }> {
 		return new Promise((resolve, reject) => {
 			const img = Image()
-			img.onload = () => resolve(img)
+			img.onload = () => {
+				document.body.appendChild(img)
+				// Sigh, Firefox doesn't have naturalWidth or naturalHeight for SVGs. :-/
+				// We have to attach to dom and use clientWidth/clientHeight.
+				const dimensions = {
+					w: img.naturalWidth || img.clientWidth,
+					h: img.naturalHeight || img.clientHeight,
+				}
+				document.body.removeChild(img)
+				resolve({ ...dimensions, image: img })
```

Comment:
is it worth checking for `naturalWidth` before adding to the dom?

Reply:
good call 👍